source/server-discovery-and-monitoring/tests/unified/closeInUse-pool-clear.json

```diff
@@ -0,0 +1,370 @@
+              {
+                "collection": {
+                  "id": "collection",
+                  "database": "database",
+                  "collectionName": "closeInUse"
+                }
+              },
+              {
+                "thread": {
+                  "id": "thread1"
+                }
+              }
+            ]
+          }
+        },      
+        {
+          "name": "insertOne",
+          "object": "collection",
+          "arguments": {
+            "document": { "_id" : 1 }
+          }
+        },
+        {
+          "name": "runOnThread",
+          "object": "testRunner",
+          "arguments": {
+            "thread": "thread1",
+            "operation": {
+                "name": "find",
+                "object": "collection",
+                "arguments": {
+                    "filter": { "$where": "sleep(5000) || true" }
+                },
+                "expectError": {
+                  "isError": true
+                }
+            }
+          }
+        },
+        {
+          "name": "failPoint",
+          "object": "testRunner",
+          "arguments": {
+            "failPoint": {
+              "configureFailPoint": "failCommand",
+              "mode": {
+                "times": 2
+              },
+              "data": {
+                "failCommands": [
+                  "hello", "isMaster"
+                ],
+                "blockConnection": true,
+                "blockTimeMS": 1500,
+                "appName": "closeInUse"
+              }
+            },
+            "client": "setupClient"
+          }
+        },
+        {
+            "name": "waitForThread",
+            "object": "testRunner",
+            "arguments": {
+              "thread": "thread1"
+            }
+        }
+      ],
+      "expectEvents": [
+        {
+          "client": "client",
+          "eventType": "command",
+          "events": [
+            {
+              "commandStartedEvent": {
+                "commandName": "insert"
+              }
+            },
+            {
+              "commandSucceededEvent": {
+                "commandName": "insert"
+              }
+            },
+            {
+              "commandStartedEvent": {
+                "commandName": "find"
+              }
+            },
+            {
+              "commandFailedEvent": {
+                "commandName": "find"
+              }
+            }
+         ]
+       },
+       {
+          "client": "client",
+          "eventType": "cmap",
+          "events": [
+            {
+              "connectionCheckedOutEvent": { }
+            },
+            {
+              "connectionCheckedInEvent": { }
+            },
+            {
+              "connectionCheckedOutEvent": { }
+            },
+            {
+              "poolClearedEvent": {
+                "closeInUseConnections": true
+              }
+            },
+            {
+              "connectionClosedEvent": { }
```

Comment:
>connectionCheckedInEvent
>connectionClosedEvent

at least in c#, we have different events order here. I would suggest either using `AssertsEventCount` step or gather more statistic which events order used in other drivers. In worst cases we can apply manual workaround this test in c#

Reply:
done